crates/typst-library/src/compute/calc.rs

```diff
@@ -188,6 +189,34 @@ pub fn sqrt(
     Ok(value.v.float().sqrt())
 }
 
+/// Calculates the nth root of a number.
+///
+/// ```example
+/// #calc.root(4, 16) \
+/// #calc.root(3, 27)
+/// ```
+#[func]
+pub fn root(
+    /// Which root of the radicand to take
+    index: Spanned<i64>,
+    /// The expression to take the root of
+    radicand: Spanned<Num>,
```

Comment:
I think `Num` could be replaced by `f64` since no special handling of ints is necessary